src/dwarf.rs

```diff
@@ -966,6 +1022,103 @@ impl DwarfResolver {
     }
 }
 
+struct SymbolInfo {
+    name: String,
+    address: u64,
+}
+
+fn parse_die_subprogram(
+    die: &mut debug_info::DIE<'_>,
+    str_data: &[u8],
+) -> Result<Option<SymbolInfo>, Error> {
+    let mut addr: Option<u64> = None;
+    let mut name_str: Option<&str> = None;
+    for (name, _form, _opt, value) in die {
+        match name {
+            constants::DW_AT_linkage_name | constants::DW_AT_name => {
+                if name_str.is_some() {
+                    continue;
+                }
+                match value {
+                    debug_info::AttrValue::Unsigned(str_off) => {
+                        match extract_string(&str_data, str_off as usize) {
+                            Some(s) => {
+                                name_str = Some(s);
+                            }
+                            _ => {
+                                return Err(Error::new(
+                                    ErrorKind::InvalidData,
+                                    "fail to extract the name of a subprogram",
+                                ));
+                            }
+                        }
+                    }
+                    debug_info::AttrValue::String(s) => {
+                        name_str = Some(s);
+                    }
+                    _ => {
+                        return Err(Error::new(
+                            ErrorKind::InvalidData,
+                            "fail to parse DW_AT_linkage_name {}",
+                        ));
+                    }
+                }
+            }
+            constants::DW_AT_lo_pc => match value {
```

Comment:
it's actually quite often important to know where function ends as well, so probably makes sense to parse hi_pc (or whatever the name)

Reply:
agree